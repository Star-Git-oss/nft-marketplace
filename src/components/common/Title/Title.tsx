import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
import colors from '../ui-library/styles/colors';
import breakpoints from '../ui-library/styles/breakpoints';

const shared = (): FlattenSimpleInterpolation => css`
  font-family: Montserrat;
  font-style: normal;
  color: ${colors.dark_blue};
  font-weight: 700;
`;

export const H1 = styled.h1`
  ${shared};
  font-size: 60px;
`;

export const H2 = styled.h2`
  ${shared};
  font-size: 50px;
`;

export const H3 = styled.h3`
  ${shared};
  font-size: 22px;
  @media only screen and (max-width: ${breakpoints.smallMaxPx}) {
    font-size: 16px;
  }
`;

export const H4 = styled.h4`
  ${shared};
  font-size: 24px;
  margin: 0 0 0 20px;
  padding:0;
`;

export const H5 = styled.h5`
  ${shared};
  font-size: 22px;
  
`;

export const P = styled.p`
  ${shared};
  font-size: 16px;
`;

export const SubTitle = styled.p`
  color: ${colors.sub_label_gray};
  font-size: 12px;
  margin-top: 5px;
`;

export const GradientText = styled.span`
    background-color: #f3ec78;
    background-image: ${colors.purple_gradient};
    background-size: 100%;
    -webkit-background-clip: text;
    -moz-background-clip: text;
    -webkit-text-fill-color: transparent; 
    -moz-text-fill-color: transparent;
`;