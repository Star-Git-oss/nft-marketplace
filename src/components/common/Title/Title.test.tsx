/* eslint-disable no-console */
import React from 'react';
import renderer from 'react-test-renderer';
import { H1, H2, H3, H4, H5, SubTitle, GradientText } from './Title';


// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

describe('Shapes snapshots', () => {

  test('H1 renders correctly', () => {
    const tree = renderer
      .create(<H1>test</H1>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('H2 renders correctly', () => {
    const tree = renderer
      .create(<H2>test</H2>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('H3 renders correctly', () => {
    const tree = renderer
      .create(<H3>test</H3>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('H4 renders correctly', () => {
    const tree = renderer
      .create(<H4>test</H4>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('H5 renders correctly', () => {
    const tree = renderer
      .create(<H5>test</H5>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('SubTitle renders correctly', () => {
    const tree = renderer
      .create(<SubTitle>test</SubTitle>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('GradientText renders correctly', () => {
    const tree = renderer
      .create(<GradientText>test</GradientText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  
  


});