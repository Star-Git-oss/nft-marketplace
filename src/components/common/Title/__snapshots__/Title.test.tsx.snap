// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapes snapshots GradientText renders correctly 1`] = `
<span
  className="sc-pFZIQ dGHhdD"
>
  test
</span>
`;

exports[`Shapes snapshots H1 renders correctly 1`] = `
<h1
  className="sc-dlfnbm jnRgnW"
>
  test
</h1>
`;

exports[`Shapes snapshots H2 renders correctly 1`] = `
<h2
  className="sc-hKgILt cxTFPj"
>
  test
</h2>
`;

exports[`Shapes snapshots H3 renders correctly 1`] = `
<h3
  className="sc-eCssSg dNNVqm"
>
  test
</h3>
`;

exports[`Shapes snapshots H4 renders correctly 1`] = `
<h4
  className="sc-jSgupP cDHmtz"
>
  test
</h4>
`;

exports[`Shapes snapshots H5 renders correctly 1`] = `
<h5
  className="sc-gKsewC fmoNTL"
>
  test
</h5>
`;

exports[`Shapes snapshots SubTitle renders correctly 1`] = `
<p
  className="sc-fubCfw kZRfsJ"
>
  test
</p>
`;
